### ISSUE TITLE
Check segment count before we force merge in ILM

### ISSUE THREAD

**Issue by gmarouli** [elastic/elasticsearch#93973]
### Description

As a user of ILM, I would like to take all the opportunities to force merge my index without overwhelming my cluster with force merge requests since they are expensive (https://github.com/elastic/elasticsearch/issues/86013).

Currently, if I have the following policy:
```
  {
    "policy" : {
      "phases" : {
        "hot" : {
          "min_age" : "0ms",
          "actions" : {
            "rollover" : {
              "max_primary_shard_size" : "50gb",
              "max_age" : "7d",
              "min_docs" : 1
            },
             "forcemerge" : {
              "max_num_segments" : 1
            }
          }
        },
        "warm" : {
          "min_age" : "12h",
          "actions" : {
            "set_priority" : {
              "priority" : 50
            }
          }
        },
        "cold" : {
          "min_age" : "13h",
          "actions" : {
            "allocate" : {
              "number_of_replicas" : 0,
              "include" : { },
              "exclude" : { },
              "require" : { }
            },
            "searchable_snapshot" : {
              "snapshot_repository" : "found-snapshots",
              "force_merge_index" : true
            },
            "set_priority" : {
              "priority" : 0
            }
          }
        }
      }
    }
```
Then force merge will be executed as follows:
- The index will be force merged on the hot tier successfully
- The index will be force merged on the warm tier, mainly it will be added to the queue and then it will be no-op
Or
- The index will not be force merged on the hot tier successfully
- The index will be force merged on the warm tier (hopefully) successfully this time.

The second force merge gives my index one more opportunity to be force merged and be in the best state to become a searchable snapshot. On the other hand, since the force merge is an expensive operation it can clog my warm tier.

The request is: can we possibly check the number of segments in `ILM` and depending on the check decide if we will force merge or not.

Note: this enhancement request is not necessary if https://github.com/elastic/elasticsearch/issues/86013 gets implemented.

**Comment by DaveCTurner** [comment]
I would much prefer to do this sort of optimisation within the force-merge action itself. As noted in #86013, there are some subtleties that make it hard to get this kind of thing completely right, but I worry that ILM is the wrong place to be implementing much of this sort of logic.

If we can't resolve #86013 completely then maybe we can find some other improvements in this general area. For instance, if ILM really only wants to do something for segment counts greater than 1 (in particular, it's not changing the codec) then perhaps we can add a parameter to the force-merge action to trigger an up-front check for no-op merges based on this parameter.

**Comment by dakrone** [comment]
+1 to what David said, I would prefer to solve this in #86013.

**Comment by gmarouli** [comment]
I completely agree, great to see that https://github.com/elastic/elasticsearch/issues/86013 is getting more attention!
